feat: Added tx-new command supporting interactive mode, token transfers, and custom gas settings for better transaction management.

pranavkonde opened the pull request:
# Enhanced Transaction Command

## Description
Added a new `tx-new` command that provides enhanced transaction capabilities for the RSK CLI.

### New Features
- Interactive mode for guided transaction creation
- Support for RBTC and token transfers
- Advanced gas management options
- Transaction simulation before sending
- Detailed transaction feedback and explorer links

### Usage Examples
```bash
# Interactive Mode
rsk-cli tx-new --testnet

# Direct RBTC Transfer
rsk-cli tx-new --testnet --address 0x... --value 0.001

# Token Transfer
rsk-cli tx-new --testnet --address 0x... --value 1 --token 0x...

# Advanced Transfer with Custom Gas
rsk-cli tx-new --testnet --address 0x... --value 0.001 --gas-limit 21000 --max-fee 0.00001
```

chrisarevalo11 commented:
Hi @pranavkonde!

Firstly I want to thank you for your contribution! 

I find interesting what you are proposing in this contribution, specifically interactive mode, token transfer and custom gas. Good work defining these features. However, I'd prefer for you to update the existing transfer command and add the features you're proposing, they're looking pretty well.

Let me know if you need any help, will be more than happy to help

pranavkonde commented:
> Hi @pranavkonde!
> 
> Firstly I want to thank you for your contribution!
> 
> I find interesting what you are proposing in this contribution, specifically interactive mode, token transfer and custom gas. Good work defining these features. However, I'd prefer for you to update the existing transfer command and add the features you're proposing, they're looking pretty well.
> 
> Let me know if you need any help, will be more than happy to help

Hi! Thank you for your feedback and guidance.

I've made the suggested changes by merging the functionality into the existing `transfer` command. Here's what I've done:

1. Enhanced the `transfer` command with:
   - Interactive mode support (`-i, --interactive`)
   - Custom gas parameters (`--gas-limit`, `--priority-fee`)
   - Custom transaction data support (`--data`)
   - Removed the redundant `tx-new` command

2. Fixed the EIP-1559 fee support issue for Rootstock network by:
   - Removing `maxFeePerGas` parameter
   - Using standard gas price mechanism
   - Supporting `priority-fee` for custom gas pricing

The functionality remains the same but is now more intuitive and consolidated into a single command. Users can now use:

```bash
# Interactive Mode
rsk-cli transfer --testnet -i

# Direct RBTC Transfer
rsk-cli transfer --testnet --address 0x... --value 0.001

# Token Transfer
rsk-cli transfer --testnet --address 0x... --value 1 --token 0x...

# Advanced Transfer with Custom Gas
rsk-cli transfer --testnet --address 0x... --value 0.001 --gas-limit 21000 --priority-fee 0.000001
```

Please review the changes and let me know if any adjustments are needed.

chrisarevalo11 commented:
Hey @pranavkonde!

Quick question over here: could you provide an example testing the transfer command making use of `data` and `gas-price` options, please?

pranavkonde commented:
> Hey @pranavkonde!
> 
> Quick question over here: could you provide an example testing the transfer command making use of `data` and `gas-price` options, please?

Hey @chrisarevalo11!

Sure! Here’s an example of how to use the transfer command with both data and gas-price options in the RSK CLI:

1. **Transfer Command** (without data):
   ```bash
   rsk-cli transfer --testnet --address 0xa5f45f5bddefC810C48aCC1D5CdA5e5a4c6BC59E --value 0.000001 --gas-price 0.0000000001
   ```
![Screenshot 2025-06-25 203213](https://github.com/user-attachments/assets/c6431b1a-1afb-4c93-b497-e8fa664614d1)

2. **Transfer Command with Data**:
   ```bash
   rsk-cli transfer --testnet --address 0xa5f45f5bddefC810C48aCC1D5CdA5e5a4c6BC59E --value 0.000001 --gas-price 0.0000000001 --data 0xabcdef1234567890
   ```
![Screenshot 2025-06-25 203417](https://github.com/user-attachments/assets/bb80b050-bf24-4de1-a01c-631132df2ca2)

In the second command, the `--data` option allows you to include additional data (in this case, `0xabcdef1234567890`) with the transaction. This can be useful for smart contract interactions or other purposes where you need to send extra information along with the transfer.

Let me know if you need any more examples or further clarification!

chrisarevalo11 commented:
Hi @pranavkonde!

Thanks a lot for the examples, it is a lot more clear. 

They seem to be working correctly! One last thing before proceeding with the approval: please update the readme containing the documentation of your updates!

Let me know if you need any help

ezequiel-rodriguez commented:
Hey @pranavkonde, 
Thank you for your contribution — it looks very thorough!

I haven’t had the chance to go through the entire PR in detail yet, as it's quite extensive, but I wanted to share some early feedback:

* I noticed that in several commands, the output is missing emojis.  For example, in src/utils/thirdwebHelper.ts you're using emojis.  It's important to follow the existing convention across commands and include emojis where appropriate.

* Currently, only ERC-20 and ERC-721 contracts are supported for deployment.  It would be great if you could add support for deploying arbitrary contract types as well.

Thanks again! 
Looking forward to reviewing the rest in more detail.

pranavkonde commented:
> Hi @pranavkonde!
> 
> Thanks a lot for the examples, it is a lot more clear.
> 
> They seem to be working correctly! One last thing before proceeding with the approval: please update the readme containing the documentation of your updates!
> 
> Let me know if you need any help

Hi @chrisarevalo11 

Thank you for your feedback! I have updated the README to include the documentation of the recent updates. Please take a look and let me know if everything looks good or if you need any further changes.

I appreciate your support!

ezequiel-rodriguez commented:
Thank you for your contribution @pranavkonde. 
Someone will contact you shortly regarding the payment for the Hacktivator Program.

Cheers
